README.rst

```diff
@@ -0,0 +1,323 @@
+
+Xen guest domains binary images
+*******************************
+
+The **zephyr-dom0-xt** takes pre-built Xen guest domains images from ``prebuilt/`` folder and
+they are integrated directly into Zephyr **zephyr-dom0-xt**. Now only two sets of Xen guest images
+is supported which names are defined by following Kconfig options:
+
+- **CONFIG_DOM_CFG_DOMD_IMAGE_BIN_FILE**: Location for Domain-D kernel binary.
+  For RPI 5 it is ``zephyr_blinky.bin`` corresponding to Zephyr ``samples/basic/blinky`` sample.
+- **CONFIG_DOM_CFG_DOMD_DTB_BIN_FILE**: Location for Domain-D device-tree binary.
+- **CONFIG_DOM_CFG_DOMU1_IMAGE_BIN_FILE**: Location for Domain-U kernel binary
+  For RPI 5 it is ``zephyr_sync.bin`` corresponding to Zephyr ``samples/synchronization`` sample.
+- **CONFIG_DOM_CFG_DOMU1_DTB_BIN_FILE**: "Location for Domain-U device-tree"
+
+To use custom binary images they shell be copied into ``prebuilt/`` folder.
+
+This approach is not very convenient and in todo list.
+
+Building and Running
+********************
+
+The **zephyr-dom0-xt** can be built for multiple supported boards.
+
+For example, to build it for **rpi_5** board with one Xen guest domain configuration
+(DomD, configuration in ``src/dom_cfg/rpi_5_domd.c``) run below command:
+
+.. code-block:: bash
+
+    west build -b rpi_5 -p always -S xen_dom0 zephyr-dom0-xt -- \
+    -DCONFIG_USE_DEFAULT_DOM_CFG=n -DCONFIG_DOM_CFG_BOARD_EXT=\"domd\"
+
+For example, to build it for **rpi_5** board with two Xen guest domain configurations
+(DomD/DomU, configuration in ``src/dom_cfg/rpi_5.c``) run below command:
+
+.. code-block:: bash
+
+    west build -b rpi_5 -p always -S xen_dom0 zephyr-dom0-xt -- \
+    -DCONFIG_USE_DEFAULT_DOM_CFG=n -DCONFIG_DOM_CFG_NUMBER=2
+
+Running
+=======
+
+The **zephyr-dom0-xt** application shell be started by `XEN`_ as Dom0 domain.
+This process is out of scope of this document.
+
+For reference use:
+`Zephyr boot under XEN from u-boot as Dom0 <https://github.com/xen-troops/meta-xt-rpi5/wiki/RPI-5-Zephyr#zephyr-boot-under-xen-from-u-boot-as-dom0>`_
+
+RPI5 Dom0 run example with two domains configuration
+****************************************************
+
+The **zephyr-dom0-xt** shell be build with already prepared Xen guest domain configuration file
+``src/dom_cfg/rpi_5.c`` for RPI5 which contains two Xen guest domain configurations:
+
+- **domd_cfg_0**: It mockups DomD behavior and uses real RPI5 GPIO HW ``/soc/gpio@7d517c00``.
+  It's intended to be used with Zephyr Blinky sample.
+- **domd_cfg_1**: It mockups DomU which pure Xen guest domain without using real RPI5 HW.
+  It's intended to be used with Zephyr samples/synchronization sample.
+
+.. note::
+
+    For RPI5 It's default configuration, so this step can be skipped.
+
+Prepare Xen guest domains binaries
+==================================
+
+Build Zephyr Blinky sample for Zephyr **xenvm** board with RPI5 snippet ``rpi_5_xen_domd``:
```

Comment:
Please add links to Zephyr branches/repo

Reply:
I'll prepare follow up patch for this to use non personal repo.